cpp/ps0004/main.cpp

```diff
@@ -4,24 +4,26 @@ using namespace std;
 
 int main()
 {
-    int num  ;
-    char character  ;
-    cout << "Enter a number from 32 to 126 :  \n" ;
-    cin >> num  ;
-
-    if ( (float)num )
-        cout <<"Invalid: a floating point number \n";
-    else if (  num <32 || num > 126  )
-        cout <<"Invalid: number is out of the range! \n " ;
-    else  if (  num = character )
-        cout <<"Invalid character/s or zero \n " ;
-
-
+    // prompting user for number 1/1.5
+    float num;
+    char character;
+    cout << "Enter a number from 32 to 126 :  \n";
+    cin >> num;
+
+    // checking invalidaty of user's input 1.5/4
+    // print the character 1.5/2
+    if (num == 0)
+        cout <<"Invalid character/s or zero \n "; // 0.25/1
+    else if (num != (int)num)
+        cout << "Invalid: a floating point number \n"; // 0.25/1
+    else if (num < 32 || num > 126 )
+        cout << "Invalid: number is out of the range! \n "; // 1/1
+    else
+    {
+        character = (char)num;
         cout <<"the character for " << num << " is " << character << ". \n" ;
-
-
-
-
+    }
 
     return 0;
 }
+// good work 👍 5/10

```

Comment:
- [x] program is in the correct path
- [ ] no syntax errors, bugs or warnings while compilation and execution
- [x] formatted code to be clear and readable
- [ ] commented some phrases for blocks of code
- [ ] main returns non-zero/zero number according to the case

you can check the [problem solution](https://onlinegdb.com/9x6wLTlkK) for more